datumaro/components/hl_ops.py

```diff
@@ -0,0 +1,192 @@
+
+    Returns: a lazy-computible wrapper around the input dataset
+    """
+
+    if isinstance(method, str):
+        if env is None:
+            env = Environment()
+        method = env.transforms[method]
+
+    produced = method(dataset, **kwargs)
+
+    return Dataset(source=produced, env=env)
+
+def filter(dataset: IDataset, expr: str, *, #pylint: disable=redefined-builtin
+        filter_annotations: bool = False,
+        remove_empty: bool = False) -> IDataset:
+    """
+    Filters-out some dataset items or annotations, using a custom filter
+    expression.
+
+    Args:
+        dataset (IDataset) - The dataset to be filtered
+        expr (str) - XPath-formatted filter expression
+            (e.g. "/item[subset = 'train']", "/item/annotation[label = 'cat']")
+        filter_annotations (bool) - Indicates if the filter should be
+            applied to items or annotations
+        remove_empty (bool) - When filtering annotations, allows to
+            exclude empty items from the resulting dataset
+
+    Returns: a lazy-computible wrapper around the input dataset
+    """
+
+    if filter_annotations:
+        return transform(dataset, XPathAnnotationsFilter,
+            xpath=expr, remove_empty=remove_empty)
+    else:
+        if not expr:
+            return dataset
+        return transform(dataset, XPathDatasetFilter, xpath=expr)
+
+def merge(*datasets: IDataset) -> IDataset:
+    """
+    Merges few datasets using the "simple" (exact matching) algorithm:
+
+    - items are matched by (id, subset) pairs
+    - matching items share the data available (nothing + nothing = nothing,
+        nothing + something = something, something A + something B = conflict)
+    - annotations are matched by value and shared
+    - in case of conflicts, throws an error
+
+    Returns: a lazy-computible wrapper around the input datasets
+    """
+
+    categories = ExactMerge.merge_categories(d.categories() for d in datasets)
+    return DatasetItemStorageDatasetView(ExactMerge.merge(*datasets),
+        categories=categories)
+
+def run_model(dataset: IDataset, model: Union[ModelTransform, Launcher], *,
+        batch_size: int = 1, **kwargs) -> IDataset:
+    """
+    Applies a model to dataset items' media and produces a dataset with
+    media and annotations.
+
+    Args:
+        dataset (IDataset) - The dataset to be transformed
+        model (Launcher or Modeltransform) - The model to be
+            applied to the dataset.
+        batch_size (int) - The number of dataset items, processed
+            simultaneously by the model
+        **kwargs - Optional parameters for the model
+
+    Returns: a lazy-computible wrapper around the input dataset
+    """
+
+    if isinstance(model, Launcher):
+        return transform(dataset, ModelTransform, launcher=model,
+            batch_size=batch_size, **kwargs)
+    elif isinstance(model, ModelTransform):
+        return transform(model, batch_size=batch_size, **kwargs)
```

Comment:
This branch is suspicious - it doesn't involve the input dataset at all.

Reply:
Fixed